listenbrainz/webserver/static/js/src/UserArtists.tsx

```diff
@@ -114,6 +98,42 @@ export default class UserArtists extends React.Component<
     return result;
   };
 
+  handlePageChange = async (page: number): Promise<void> => {
+    const { user } = this.props;
+    const offset = (page - 1) * this.ROWS_PER_PAGE;
+    try {
+      const data = await this.getData(user.name, offset);
+      this.currPage = page;
+      this.setState({ data: this.processData(data, offset) });
+    } catch (error) {
+      this.handleError(error);
+    }
+  };
+
+  handleError = (error: Error): void => {
+    // Error Boundaries don't catch errors in async code.
+    // Throwing an error in setState fixes this.
+    // This is a hacky solution but should be fixed with upcoming concurrent mode in React.
+    const { user } = this.props;
+
+    if (
+      error.name === "SyntaxError" &&
+      error.message === "Unexpected end of JSON input"
+    ) {
+      // If the above error is thrown, it means stats haven't been calculated, show an
+      // appropriate message to the user.
+      this.setState(() => {
+        throw new Error(
+          `Statistics for user: ${user.name} have not been calculated yet. Please try again later.`
+        );
+      });
+    }
+
+    this.setState(() => {
+      throw error;
+    });
+  };
+
```

Comment:
Not sure what's happening here. Are you setting the state to an error? how does that work? Is it something to do with the ErrorBoundary component?

Reply:
`Error Boundary` doesn't catch errors in `async` functions. Throwing an error in `setState` fixes this issue. More information about his can be found [here](https://medium.com/trabe/catching-asynchronous-errors-in-react-using-error-boundaries-5e8a5fd7b971)